### PR TITLE
chore: Add Sentry/HybridSDK as dependency

### PR DESCRIPTION
**Pull request by brustolin**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring

## :scroll: Description
Add a dependency to "Sentry/HybridSDK" in the podspec to be able to use `PrivateSentrySDKOnly.h`.

This pr should be merged only after the next cocoa release.

## :bulb: Motivation and Context

https://github.com/getsentry/team-mobile/issues/66

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] All tests passing
- [x] No breaking changes

_#skip-changelog_

### CODE REVIEW

File: RNSentry.podspec [hunk 1]
```diff
@@ -17,7 +17,7 @@ Pod::Spec.new do |s|
   s.preserve_paths = '*.js'
 
   s.dependency 'React-Core'
-  s.dependency 'Sentry', '7.30.0'
+  s.dependency 'Sentry/HybridSDK', '7.31.2'
```
Comment (brustolin) [inline RNSentry.podspec:20]:
```suggestion
  s.dependency 'Sentry/HybridSDK', '7.30.2'
```

File: RNSentry.podspec [hunk 2]
```diff
@@ -17,7 +17,7 @@ Pod::Spec.new do |s|
   s.preserve_paths = '*.js'
 
   s.dependency 'React-Core'
-  s.dependency 'Sentry', '7.30.0'
+  s.dependency 'Sentry/HybridSDK', '7.30.2'
```
Comment (marandaneto) [inline RNSentry.podspec:20]: Add changelog entry for the SDK upgrade.

Comment (brustolin) [reply RNSentry.podspec:20]: This is an internal change users don't need to know about.
We don't add changelog entries for cocoa when this is the case.
Should I do it here?

Comment (marandaneto) [reply RNSentry.podspec:20]: It's not about the use of HybridSDK header, but rather about the SDK upgrade as per my comment above.
https://github.com/getsentry/sentry-cocoa/releases/tag/7.30.2
There are more fixes in this release that affects this SDK, so it should go under the changelog.

**Comment by krystofwoldrich** [comment]
@marandaneto It's just CI, it's failing on all PRs.

**Review by krystofwoldrich — APPROVED** [review]

**Pull request merged**